src/Layout.tsx

```diff
@@ -1,36 +1,58 @@
 import React, { useState } from "react";
-import { Box, Toolbar, IconButton, Drawer, useTheme, useMediaQuery } from "@mui/material";
-import MenuIcon from '@mui/icons-material/Menu';
+import {
+  Box,
+  Toolbar,
+  IconButton,
+  Drawer,
+  useTheme,
+  useMediaQuery,
+} from "@mui/material";
+import MenuIcon from "@mui/icons-material/Menu";
 import { Outlet } from "react-router-dom";
 import { StyledAppBar } from "./components/styled/styles";
+import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
 import Routes from "./components/Routes";
-
+import uaFlag from "./img/ua.png";
+import enFlag from "./img/en.png";
+import { useThemeCustom } from "./theme/ThemeContext";
+import { useLanguage } from "./components/LanguageContext";
+import { MaterialUISwitch } from "./components/styled/Switch";
 const Layout: React.FC = () => {
+  const { darkMode, toggleDarkMode } = useThemeCustom();
+  const { language, toggleLanguage } = useLanguage();
   const [isDrawerOpen, setIsDrawerOpen] = useState(false);
   const theme = useTheme();
-  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+  const isMobileOrTablet = useMediaQuery(theme.breakpoints.between("xs", "xl"));

```

Comment:
In this case the mobile menu will be displayed on the desktop screen too, you should leave sm breakpoint in that point